contracts/libraries/VaultLifecycle.sol

```diff
@@ -406,13 +406,20 @@ library VaultLifecycle {
      * It burns oTokens from the most recent vault opened by the contract. This assumes that the contract will
      * only have a single vault open at any given time.
      * @param gammaController is the address of the opyn controller contract
-     * @param amount is the amount of otokens to burn
+     * @param currentOption is the address of the current option
      * @return amount of collateral redeemed by burning otokens
      */
-    function burnOtokens(address gammaController, uint256 amount)
+    function burnOtokens(address gammaController, address currentOption)
         external
         returns (uint256)
     {
+        uint256 numOTokensToBurn =
+            IERC20(currentOption).balanceOf(address(this));
+
+        if (numOTokensToBurn < 0) {
+            return 0;
+        }
```

Comment:
This needs to be `numOTokensToBurn == 0`, maybe we should require instead?

Reply:
sure